csm/conf/setup.py

```diff
@@ -271,6 +275,60 @@ def _config_user_permission_unset(self, bundle_path):
         Setup._run_cmd("rm -rf " + bundle_path)
         Setup._run_cmd("rm -rf " + const.CSM_PIDFILE_PATH)
 
+    def _fetch_ldap_root_password(self):
+        Log.info("Fetching LDAP root user password from Conf Store.")
+        try:
+            ldap_root_secret = Conf.get(const.CONSUMER_INDEX, self.conf_store_keys[const.KEY_ROOT_LDAP_SECRET])
+            cluster_id = Conf.get(const.CONSUMER_INDEX, self.conf_store_keys[const.KEY_CLUSTER_ID])
+            cipher_key = Cipher.generate_key(cluster_id,
+                        Conf.get(const.CSM_GLOBAL_INDEX, "CSM>password_decryption_key"))
+        except KvError as error:
+            Log.error(f"Failed to Fetch keys from Conf store. {error}")
+            return None
+        except Exception as e:
+            Log.error(f"{e}")
+            return None
+        try:
+            ldap_root_decrypted_value = Cipher.decrypt(cipher_key,
+                                                ldap_root_secret.encode("utf-8"))
+            return ldap_root_decrypted_value.decode('utf-8')
+        except CipherInvalidToken as error:
+            Log.error(f"Decryption for LDAP root user password Failed. {error}")
+            raise CipherInvalidToken(f"Decryption for LDAP root user password Failed. {error}")
```

Comment:
We already have decryption logic available in CSM code ... Use it. instead of re-writing it
Check. `Security.decrypt_conf()`